pkg/cmd/repo/archive/archive.go

```diff
@@ -29,21 +34,29 @@ func NewCmdArchive(f *cmdutil.Factory, runF func(*ArchiveOptions) error) *cobra.
 	}
 
 	cmd := &cobra.Command{
-		DisableFlagsInUseLine: true,
-
-		Use:   "archive <repository>",
+		Use:   "archive [<repository>]",
 		Short: "Archive a repository",
-		Long:  "Archive a GitHub repository.",
-		Args:  cmdutil.ExactArgs(1, "cannot archive: repository argument required"),
+		Long: heredoc.Doc(`Archive a GitHub repository.
+
+With no argument, archives the current repository.`),
+		Args: cobra.MaximumNArgs(1),
 		RunE: func(cmd *cobra.Command, args []string) error {
-			opts.RepoArg = args[0]
+			if len(args) > 0 {
+				opts.RepoArg = args[0]
+			}
+			opts.BaseRepo = f.BaseRepo
+
+			if !opts.Confirmed && !opts.IO.CanPrompt() {
+				return cmdutil.FlagErrorf("could not prompt: confirmation with prompt or --confirm flag required")
```

Comment:
What do you think about 
```suggestion
				return cmdutil.FlagErrorf("--confirm required when not running interactively")
```